### DISCUSSION TITLE
MSYS2でASP3用configure.rbが動かない

### DISCUSSION THREAD

**Discussion by Ryohsuke-Takeuchi** [toppers/users-forum#77] [Q&A]
前のディスカッションで、コンフィギュレータのruby3.0で不具合ありの投稿がありましたが、
MSYS2でNucleo簡易パッケージ（asp3_nucleo_f401re_gcc-202211228.zip）を動かそうとしましたが、
configure.rb自体が、3.0では動かないようです。
cygwinでの実行画面：
![image](https://user-images.githubusercontent.com/91311953/176134354-fa5473fa-a207-473b-9117-71c7e48dff86.png)

MSYS2での実行画面：
![image](https://user-images.githubusercontent.com/91311953/176134250-cbd928b0-13bc-484a-98fa-d3c5c19c777e.png)

**Comment by hiroakitakada** [comment]
configure.rbの中身というより，require文でエラーになっているようですね。configure.rbの47行目に何が書いてあるか，お知らせくださると幸いです（バージョンによって行が移動している可能性があるため）。もしその行が「require "fileutils"」でしたら，fileutilsのロードに失敗しているものと思われます。対処案として，47行目を削除し，下の方にある「FileUtils.move」を「File.rename」に変更して試してもらえないでしょうか？（moveだけであればFileUtilsは必要ないのに，無駄に使っているようです）

**Reply by Ryohsuke-Takeuchi** [reply]
47行目は require "shell" でshellが呼べないエラーですかね。
その下の行は、コメントで、110行が変数定義です。
![image](https://user-images.githubusercontent.com/91311953/176188091-d559988b-f5c5-4a35-9621-2e2593d4459b.png)
![image](https://user-images.githubusercontent.com/91311953/176188282-73a72830-c7af-4fba-962a-4ea15be2dcf0.png)

**Reply by Ryohsuke-Takeuchi** [reply]
require "shell" を削除すれば、以下のエラーです
![image](https://user-images.githubusercontent.com/91311953/176199853-8c2dd945-8b9a-4eeb-8b5d-a158d03bcf12.png)

**Reply by hiroakitakada** [reply]
「require "shell"」を使っているのは，ちょっと古いバージョンですね。
ASP 3.6.0のconfigure.rbは「require "shell"」を使っていませんが，さらに「require "fileutils"」も使わないバージョンを添付でお送りします（zip圧縮してあります）。次のリリースではこちらに置き換えますので，こちらで試してもらえると幸いです。
[configure.rb.zip](https://github.com/toppers/users-forum/files/9006649/configure.rb.zip)

**Reply by Ryohsuke-Takeuchi** [reply]
高田先生が修正されたconfugure.rbで、sample1のワークディレクトリはできました。
![image](https://user-images.githubusercontent.com/91311953/176381351-bd30660d-8703-40e6-8e06-5a03020391e7.png)

makeでビルドすると、以下のようなエラーがでます。pass1.rbの修正はされていました。
![image](https://user-images.githubusercontent.com/91311953/176380999-f6092405-3aad-4ddb-8461-b856f7fb1577.png)

**Reply by hiroakitakada** [reply]
これはtecsgen（TECSジェネレータ）で落ちていますね。tecsgenも，古いものはruby3.0で動かないという話しがあったので，最新版（1.8.0）にしてもらった方が良さそうです。
https://www.toppers.jp/tecs.html#d-package
tecsgen-1.8.0.tgzを展開して，その中のtecsgenディレクトリ（トップディレクトリではない）をasp3/tecsgenに移動かリンクしてください。

**Comment by Ryohsuke-Takeuchi** [comment]
高田先生 お忙しいところ、申し訳、ありません。修正された。configure.rbでエラーがでなくなりました。ちなみに、cygwinのconfigure.rbをこれに変えても、問題なくMakefile等はできます。但し、その後のmakeコマンドでエラー発生です。ユーザーズ・フォーラムに記載しました。pass1.rbの修正は、入っているみたいです。報告まで。 Windows の メール から送信 差出人: Hiroaki TAKADA送信日時: 2022年6月29日 11:02宛先: toppers/users-forumCC: 竹内良輔; Author件名: Re: [toppers/users-forum] MSYS2でASP3用configure.rbが動かない (Discussion #77) 「require "shell"」を使っているのは，ちょっと古いバージョンですね。ASP 3.6.0のconfigure.rbは「require "shell"」を使っていませんが，さらに「require "fileutils"」も使わないバージョンを添付でお送りします（zip圧縮してあります）。次のリリースではこちらに置き換えますので，こちらで試してもらえると幸いです。configure.rb.zip—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>

**Comment by Ryohsuke-Takeuchi** [comment]
tecsgenを1.8.0に変えて、TECSのエラーは出なくなりましたが、
今度は、cfgのエラーが出ました。
![image](https://user-images.githubusercontent.com/91311953/176681265-ecdd0f25-6c9b-4d0f-99d1-b24f952064c9.png)

**Reply by hiroakitakada** [reply]
こちらは，cfgがruby3に対応できていないという既知の問題です。pass1.rbを以下のように修正ください。

```
    symvalCsv = CSV.open(symvalTableFileName,
						 { skip_blanks: true, skip_lines: /^#/ })
```

を以下に修正（{} 囲みを外す）。

```
    symvalCsv = CSV.open(symvalTableFileName,
						 skip_blanks: true, skip_lines: /^#/)
```

**Reply by Ryohsuke-Takeuchi** [reply]
修正の文脈を見て、{}を付けるのかと思っていました。NULCEO_F401RE_GCCは2022年1月のリリースで、{}が付いていたので修正済みと勘違いしました。cygwinで{}を外したpass1.rbでも、問題なくビルドできますね。
MSYSでも、以下のように、ビルドできました。ありがとう、ございました。

![image](https://user-images.githubusercontent.com/91311953/176808338-40fc5733-05bc-4333-8169-373537dcc279.png)

**Reply by hiroakitakada** [reply]
解決して良かったです！

**Reply by Ryohsuke-Takeuchi** [reply]
Ubunte 22.04 LTSのターミナルでも、MSIS2と同じ修正を行えば、ASP3がビルドできることを確認しました。
